openapi/foundry_v3_test.go

```diff
@@ -0,0 +1,68 @@
+package openapi
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+func TestNewFoundryFromSpecV3(t *testing.T) {
+	sampleSchema := map[string]interface{}{
+		"properties": map[string]interface{}{
+			"foo": map[string]interface{}{
+				"type": "string",
+			},
+			"bar": map[string]interface{}{
+				"type": "number",
+			},
+		},
+		"type": "object",
+	}
+
+	spec := SchemaToSpec("com.hashicorp.v1.TestCrd", sampleSchema)
+	j, err := json.Marshal(spec)
+	if err != nil {
+		t.Fatalf("Error: %+v", err)
+	}
+
+	f, err := NewFoundryFromSpecV3(j)
+	if err != nil {
+		t.Fatalf("Error: %+v", err)
+	}
+
+	if f.(*foapiv3).doc == nil {
+		t.Fail()
+	}
+	if f.(*foapiv3).doc.Components.Schemas == nil {
+		t.Fail()
+	}
+	if f.(*foapiv3).doc.Components.Schemas == nil {
```

Comment:
This seems to be a duplicate of the conditional above.

Reply:
Thanks. Good catch!